seacrowd/sea_datasets/tydiqa/tydiqa.py

```diff
@@ -0,0 +1,469 @@
+                                        "yes_no_answer": yes_no_answers,
+                                    },
+                                    "language": lang,
+                                },
+                            }
+                    else:
+                        raise ValueError(f"No configs to match {self.config.name} in primary_task")
+
+        elif "secondary_task" in self.config.name:
+            with open(filepath, encoding="utf-8") as f:
+                data = json.load(f)
+                for article in data["data"]:
+                    title = article.get("title", "").strip()
+                    for paragraph in article["paragraphs"]:
+                        context = paragraph["context"].strip()
+                        for qa in paragraph["qas"]:
+                            question = qa["question"].strip()
+                            id_ = qa["id"]
+
+                            answer_starts = [answer["answer_start"] for answer in qa["answers"]]
+                            answers = [answer["text"].strip() for answer in qa["answers"]]
+
+                            # Features currently used are "context", "question", and "answers".
+                            # Others are extracted here for the ease of future expansions.
+                            if self.config.name == "secondary_task_source":
+                                yield id_, {
+                                    "title": title,
+                                    "context": context,
+                                    "question": question,
+                                    "id": id_,
+                                    "answers": {
+                                        "answer_start": answer_starts,
+                                        "text": answers,
+                                    },
+                                }
+
+                            elif self.config.name == "secondary_task_indonesian_source":
+                                if id_.startswith("indonesian"):
+                                    yield id_, {
+                                        "title": title,
+                                        "context": context,
+                                        "question": question,
+                                        "id": id_,
+                                        "answers": {
+                                            "answer_start": answer_starts,
+                                            "text": answers,
+                                        },
+                                    }
+                            elif self.config.name == "secondary_task_seacrowd_qa":
+                                yield id_, {"id": id_, "question_id": id_, "document_id": id_, "question": question, "type": "abstractive", "choices": [], "context": context, "answer": answers, "meta": {"answer_start": answer_starts}}
```

Comment:
nit, declare the dictionary with new lines just like you did with others above. Also create functions to not dupe the implementations

Reply:
I think this is caused by the ```make check_file=seacrowd/sea_datasets/tydiqa```, fixed.